parser/parser.go

```diff
@@ -33,173 +33,151 @@ var (
 	)
 )
 
-// FuzzProcessor reads a stream of fuzzer output lines, detects failures,
-// writes logs, and captures the failing input for later logging.
-type FuzzProcessor struct {
-	// Logger used to record informational and error messages during
-	// processing.
+// FuzzOutputProcessor handles parsing and logging of fuzzing output streams,
+// detecting failures, and capturing/logging failing input data.
+type FuzzOutputProcessor struct {
+	// Logger for informational and error messages.
 	logger *slog.Logger
 
 	// Configuration settings provided by the user
 	cfg *config.Config
 
-	// Path to the directory where the input failing corpus (if any) is
-	// stored.
-	corpusPath string
+	// Directory containing the fuzzing corpus.
+	corpusDir string
 
-	// Name of the fuzz target being processed.
-	target string
+	// Name of the fuzz target under test.
+	targetName string
 
-	// Interface responsible for writing logs to the desired output.
-	logWriter LogWriter
-
-	// Tracks the state of the processing, including whether a failure was
-	// detected.
-	State *ProcessState
-}
-
-// ProcessState holds mutable state information while processing fuzzer output.
-type ProcessState struct {
-	// SeenFailure indicates whether a failure marker line has been spotted.
-	SeenFailure bool
-
-	// ErrorData contains the formatted contents of the failing testcase.
-	ErrorData string
-
-	// InputPrinted indicates whether the failing input has already been
-	// captured.
-	InputPrinted bool
+	// File handle for writing failure logs.
+	logFile *os.File
```

Comment:
Looks like we don't even need this field, since it is only used locally within `processFailureLines`.

```suggestion
```